### ISSUE TITLE
Request: a command to run a command under env

### ISSUE THREAD

**Issue by pitrou** [conda/conda#7066]
**I'm submitting a...**
  - [ ] bug report
  - [x] feature request

conda environments are easy to access from shell scripts. But it's not so nice from e.g. some Python code. Ideally, a piece of code that knows the name of an environment should be able to execute easily a command inside that environment.

Let's say I want to run `pip -V` inside a given environment named "foobar". This won't work:
```python
subprocess.check_call(['conda', 'activate', 'foobar'])
subprocess.check_call(['pip', '-V'])
```

What I'm suggesting is to be able to write something like this:
```python
subprocess.check_call(['conda', 'env', 'run', '-n', 'foobar', 'pip', '-V'])
```
... making the `pip` subcommand execute with all the right environment variables set up (such as PATH, CONDA_PREFIX, etc.).

**Comment by kalefranz** [comment]
This is sort of along the lines of bringing back `conda run`. Except `conda run` had an implementation detail where, on the command line, you’d request to run the package name instead of the executable name. That was just too whacky and limiting for me 🤷🏻‍♂️

This feature request is definitely on the radar, but we first need https://github.com/conda/conda/issues/6820

I don’t want to be invoking a shell when all we really should be doing is spawning a process, dressing it with environment variables, and exec-ing to the requested executable.

**Comment by kalefranz** [comment]
`conda run` was implemented in #7320 and will be available in the pending conda 4.6
